app/controllers/goldencobra/api/v2/articles_controller.rb

```diff
@@ -211,77 +168,89 @@ def breadcrumb
-            author = Goldencobra::Author.find_or_create_by_lastname(params[:author][:lastname])
-            new_article.author = author
+            Goldencobra::Author.find_or_create_by_lastname(
+              params[:author][:lastname]
+            )
           end
+        end
 
-          if params[:images].present?
-            params[:images].each do |i|
-              img = Goldencobra::Upload.create(i[:image])
-              article.images << img
-            end
+        def create_article(article_param)
+          new_params = setup_params(article_params)
+          new_article = Goldencobra::Article.new(new_params)
+          article_param.each do |i|
+            new_article.images << Goldencobra::Upload.create(i[:image])
           end
-
-
-          #Set externel Referee
-          new_article.external_referee_id = params[:referee_id]
-          new_article.external_referee_ip = request.env['REMOTE_ADDR']
-
-          # Try to save the article
           new_article.save
-          return new_article
+          new_article
         end
 
+        def check_for_current_user
+          render_error(nil, 403) and return unless current_user
+        end
 
-        def update_article(article_param)
-          # Input validation
-          return nil if check_presents_of_article_params(article_param) == false
-
-          # Get existing article
-          article = Goldencobra::Article.where(creator_id: current_user.id).find_by_external_referee_id(params[:referee_id])
+        def check_for_article
+          render_error("article data missing") and return unless params[:article]
+        end
 
-          # Update existing article
-          article.update_attributes(params[:article])
+        def check_for_referee
+          render_error("referee_id missing") and return unless params[:referee_id]
+        end
 
-          # Try to save the article
-          return article
+        def render_error(message, status = 400)
+          if message
+            render status: status, json: { status: status, error: message }
+          else
+            render status: status, json: { status: status }
+          end
         end
 
-        private
+        def existing_articles
+          attrs = {
+            creator_id: current_user.id,
+            external_referee_id: params[:referee_id]
+          }.delete_if { |_k, v| v.blank? }
+          Goldencobra::Article.where(attrs)
+        end
 
-        # Exit Reasons if params are not complete
-        # @param article_param [Hash] article_params
-        #
-        # @return [Boolean] True if all need params are present?
-        def check_presents_of_article_params(article_param)
-          return false unless article_param
-          return false unless params[:article]
-          return false unless current_user
-          return false unless params[:referee_id]
-          return true
+        def create_images(article, images)
+          images.each do |key,value|
```

Comment:
Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.<br>Space missing after comma.